### PR TITLE
ADD: Added filters

### PR DESCRIPTION
**Pull request by zoef-yt**
In this pr, I added 

- [x] Search feature
- [x] filtering
- [x] notes flow
- [x] Bug fix
- [x] Changes suggested in #5  and #6  

Things not to check:
Edit feature.

### CODE REVIEW

**Review by sanu29 — APPROVED** [review]
All good to proceed with commit according to me. Great work could not find any issues.
Spacing is great as well your naming conventions are meaningful and easy.
File structure is also good. Clean Code and readable.
Great work.

File: src/Components/Header/Header.jsx [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
 	const nameInitials = user?.foundUser.firstName[0] + user?.foundUser.lastName[0];
 	return (
 		<nav className='header'>
 			<NavLink className='header-brand' to='/'>
 				<h1>Jot It Down</h1>
 			</NavLink>
 
-			<input type='search' className='text-field' placeholder='Search For Games' />
+			<input
+				type='search'
```
Comment (muskanbagrecha) [inline src/Components/Header/Header.jsx:18]: Great. Did not know search type exists.

File: src/Components/NotesCard/NotesCard.jsx [hunk 2]
```diff
@@ -1,27 +1,110 @@
...
+		notesToTrash(note);
+		deleteArchiveNotes(_id);
+	};
+
+	const undoDeleteHandler = (_id, note) => {
+		deleteTrashNotes(_id);
+		postNotes(note);
+	};
+
+	// const send = (_id) => {
```
Comment (muskanbagrecha) [inline src/Components/NotesCard/NotesCard.jsx]: Remove commented code.

File: src/Components/NotesPage/FilterBar/FilterBar.css [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	grid-area: filter-bar;
+	position: -webkit-sticky;
+	position: sticky;
+	top: 0rem;
+	z-index: 1;
+	background-color: var(--colour-black-one);
+}
+
+.filter-items {
+	/* width: 60vw; */
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/FilterBar/FilterBar.css]: Remove commented code

File: src/Components/NotesPage/FilterBar/FilterBar.jsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import { useFilter } from '../../../context';
+import { priorities } from '../../../data/priorities';
+import { tags } from '../../../data/tags';
+import './FilterBar.css';
+const FilterDropDownbutton = (props) => {
+	return (
+		<div className='dropdown'>
+			<button className='btn btn-filter btn-primary dropdown-btn'>{props.buttonName}</button>
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/FilterBar/FilterBar.jsx]: You can destructure props.

File: src/Components/NotesPage/FilterBar/FilterBar.jsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+			<div style={{ transform: `translateX(-${props.scroll}px)` }} className='dropdown-content'>
+				{props.children}
+			</div>
+		</div>
+	);
+};
+
+const FilterBar = () => {
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/FilterBar/FilterBar.jsx:17]: Loved your approach to make the filter panel.

File: src/Components/NotesPage/FilterBar/FilterBar.jsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+	const { FilterState, FilterDispatch } = useFilter();
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/FilterBar/FilterBar.jsx]: Suggestion: Dont make the state and dispatch function capital as they are not components.

File: src/Components/NotesPage/FilterBar/FilterBar.jsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+									<input
+										type='radio'
+										name='sort-by-date'
+										checked={FilterState.sortByDate === 'OLDEST_TO_NEWEST'}
+										onChange={() => FilterDispatch({ type: 'SORT_BY_DATE', payload: 'OLDEST_TO_NEWEST' })}
+									/>
+									Sort Oldest to Newest
+								</label>
+
+								<label>
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/FilterBar/FilterBar.jsx:45]: You can avoid using implicit labels if possible, not all assistive technologies catch it.

File: src/Components/NotesPage/NotesPage.jsx [hunk 8]
```diff
@@ -1,80 +1,126 @@
...
+};
+
+const getDataByTag = (selectedTag, data) => (selectedTag === 'None' || selectedTag === '' ? data : data.filter((note) => note.tag === selectedTag));
+
+const getDataByPriority = (selectedPriority, data) =>
+	selectedPriority !== '' && selectedPriority !== 'All' ? data.filter((note) => note.priority == selectedPriority) : data;
+
+const getDataBySearch = (searchedWord, data) => {
+	return searchedWord !== null && searchedWord !== '' ? data.filter((note) => note.title.toLowerCase().includes(searchedWord.toLowerCase())) : data;
+};
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/NotesPage.jsx:126]: You can also add an extra trim() to ignore trailing spaces in the search keyword.

File: src/context/FilterContext/FilterContext.jsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+				...state,
+				search: action.payload,
+			};
+
+		case 'SORT_BY_DATE':
+			if (action.payload === 'OLDEST_TO_NEWEST') {
+				return {
+					...state,
+					sortByDate: 'OLDEST_TO_NEWEST',
+					dataToShow: [...state.dataToShow].sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt)),
```
Comment (muskanbagrecha) [inline src/context/FilterContext/FilterContext.jsx:52]: Appreciation: Great that you are spreading the array first as sort is not a pure function.

File: src/context/NotesContext/NotesContext.jsx [hunk 10]
```diff
@@ -1,15 +1,17 @@
...
 
 const NotesContext = createContext();
 
 const NotesProvider = ({ children }) => {
 	const [notes, setNotes] = useState([]);
 	const { response: notesResponse, operation: fetchNotes } = useAxios();
-
+	const { FilterDispatch } = useFilter();
 	useEffect(() => {
 		if (notesResponse != null && notesResponse.notes) {
```
Comment (muskanbagrecha) [inline src/context/NotesContext/NotesContext.jsx]: You can just give if (notesResponse?.notes) I think. Correct me if I'm wrong.

**Review by muskanbagrecha — CHANGES_REQUESTED** [review]
Great work on the filters. Also appreciate how you have taken care of edge cases of not showing filters if there are no notes, etc. Definitely look into suggested changes and let me know your views.

File: src/Components/NotesCard/NotesCard.jsx (hunk 2 above)
Comment (zoef-yt) [reply src/Components/NotesCard/NotesCard.jsx]: Done.

File: src/Components/NotesPage/FilterBar/FilterBar.css (hunk 3 above)
Comment (zoef-yt) [reply src/Components/NotesPage/FilterBar/FilterBar.css]: Done.

File: src/Components/NotesPage/FilterBar/FilterBar.jsx (hunk 4 above)
Comment (zoef-yt) [reply src/Components/NotesPage/FilterBar/FilterBar.jsx]: done

File: src/Components/NotesPage/FilterBar/FilterBar.jsx (hunk 6 above)
Comment (zoef-yt) [reply src/Components/NotesPage/FilterBar/FilterBar.jsx]: Done

File: src/Components/NotesPage/FilterBar/FilterBar.jsx (hunk 7 above)
Comment (zoef-yt) [reply src/Components/NotesPage/FilterBar/FilterBar.jsx:45]: Okay will have to look into it won't be changing it for this submmision.

File: src/context/NotesContext/NotesContext.jsx (hunk 10 above)
Comment (zoef-yt) [reply src/context/NotesContext/NotesContext.jsx]: you are correct. Thanks.

**Review by muskanbagrecha — APPROVED** [review]
Good to merge as you have implemented the suggested changes. Great job overall. Filters came out pretty well.

**Pull request merged**